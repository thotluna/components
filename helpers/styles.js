import classNames from 'classnames'

export const getDynamicClasses = (cssModule = {}, props = {}, classes) => {
  return classes.reduce((classesObject, classKey) => {
    const propValue = props[classKey]
    const className = cssModule[`${classKey}-${propValue}`]

    return className && propValue
      ? { ...classesObject, [className]: propValue }
      : classesObject
  }, {})
}

export const getModuleClasses = (cssModule, classKey) => {
  return (cssModule && cssModule[classKey]) || classKey
}

export const getObjectClasses = (cssModule, object) => {
  return Object.keys(object).reduce((classes, classKey) => {
    const className = cssModule[classKey]
    return className ? { ...classes, [className]: object[classKey] } : classes
  }, {})
}

export const getClasses = (cssModule = {}) => (props) => (...args) => {
  const allClass = args.map((arg) => {
    if (Array.isArray(arg)) {
      const dinamicClasses = getDynamicClasses(cssModule, props, arg)
      // console.log('>>>dinamic', dinamicClasses)
      return dinamicClasses
    } else if (typeof arg === 'string') {
      const moduleClass = getModuleClasses(cssModule, arg)
      // console.log('>>>module', moduleClass)
      return moduleClass
    } else if (typeof arg === 'object') {
      const objectClass = getObjectClasses(cssModule, arg)
      // console.log('>>>object', objectClass)
      return objectClass
    }
  })
  // console.log(allClass)

  const result = classNames(allClass)

  // console.log(result)
  return result
}
