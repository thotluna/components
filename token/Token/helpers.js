import toKebabCase from '../../utils/toKebalCase'

export const getTokenItems = (tokens, tokenKey) => (group) => {
  if (group) {
    return Object.keys(tokens[group]).map((key) => ({
      label: `--${toKebabCase(tokenKey)}-${group}-${toKebabCase(key)}`,
      value: tokens[group][key],
    }))
  }

  return Object.keys(tokens).map((key) => ({
    label: `--${toKebabCase(tokenKey)}-${toKebabCase(key)}`,
    value: tokens[key],
  }))
}
