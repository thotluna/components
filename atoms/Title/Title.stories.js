import React from 'react'
import { Title, options } from '.'
import { getVariusTemplate, getOptionsArgTypes } from '../../helpers/storybook'

const Template = (args) => <Title {...args} />
const ListTemplate = getVariusTemplate(Title)

export default {
  title: 'Atoms/Title',
  component: Title,
  args: {
    size: 'md',
    color: 'primary',
    children: 'Consent doubt Tuckborough challenge destroying.',
  },
  argTypes: {
    children: { control: 'text' },
    color: getOptionsArgTypes(options.colors),
    size: getOptionsArgTypes(options.sizes),
  },
}

export const Default = Template.bind({})

export const Sizes = ListTemplate.bind({})
Sizes.args = { items: options.sizes.map((size) => ({ size })) }

export const Colors = ListTemplate.bind({})
Colors.args = { items: options.colors.map((color) => ({ color })) }
