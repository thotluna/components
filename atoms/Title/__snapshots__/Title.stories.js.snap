// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Title Colors 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <h4
    className="title color-base"
  >
    It is a Title.
  </h4>
  <h4
    className="title color-primary"
  >
    It is a Title.
  </h4>
  <h4
    className="title color-error"
  >
    It is a Title.
  </h4>
</div>
`;

exports[`Storyshots Atoms/Title Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <h4
    className="title color-base"
  >
    It is a Title.
  </h4>
</div>
`;

exports[`Storyshots Atoms/Title Size 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <h4
    className="title color-base"
  >
    It is a Title.
  </h4>
  <h3
    className="title color-base"
  >
    It is a Title.
  </h3>
  <h2
    className="title color-base"
  >
    It is a Title.
  </h2>
  <h1
    className="title color-base"
  >
    It is a Title.
  </h1>
</div>
`;

exports[`Storyshots Atoms/Title Size Color 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <h4
    className="title color-base"
  >
    It is a Title.
  </h4>
  <h3
    className="title color-base"
  >
    It is a Title.
  </h3>
  <h2
    className="title color-base"
  >
    It is a Title.
  </h2>
  <h1
    className="title color-base"
  >
    It is a Title.
  </h1>
  <h4
    className="title color-primary"
  >
    It is a Title.
  </h4>
  <h3
    className="title color-primary"
  >
    It is a Title.
  </h3>
  <h2
    className="title color-primary"
  >
    It is a Title.
  </h2>
  <h1
    className="title color-primary"
  >
    It is a Title.
  </h1>
  <h4
    className="title color-error"
  >
    It is a Title.
  </h4>
  <h3
    className="title color-error"
  >
    It is a Title.
  </h3>
  <h2
    className="title color-error"
  >
    It is a Title.
  </h2>
  <h1
    className="title color-error"
  >
    It is a Title.
  </h1>
</div>
`;
